File: backend/purchase-service/src/main/java/com/stockpro/purchase/entity/PurchaseOrder.java
package com.stockpro.purchase.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "purchase_orders")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class PurchaseOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long supplierId;



    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private PoStatus status = PoStatus.DRAFT;

    @Column(nullable = false)
    private Long orderedBy;

    private Long approvedBy;
    private Long receivedBy;
    @Column(columnDefinition = "TEXT")
    private String notes;

    private LocalDate expectedDate;

    private LocalDateTime receivedDate;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "purchaseOrder",
            cascade = CascadeType.ALL,
            orphanRemoval = true,
            fetch = FetchType.EAGER)
    @Builder.Default
    private List<PurchaseOrderItem> items = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/core/models/purchase-order.ts
export type PoStatus = 'DRAFT' | 'APPROVED' | 'PARTIALLY_RECEIVED' | 'RECEIVED' | 'CANCELLED';
export interface PurchaseOrderItem {
  id: number; productId: number; warehouseId: number;
  quantityOrdered: number; quantityReceived: number;
  unitPrice: number; createdAt: string;
}
export interface PurchaseOrder {
  id: number; supplierId: number; status: PoStatus;
  orderedBy: number; approvedBy: number; receivedBy: number;
  notes: string; expectedDate: string; receivedDate: string;
  createdAt: string; updatedAt: string; items: PurchaseOrderItem[];
}
export interface PurchaseOrderItemRequest {
  productId: number; warehouseId: number;
  quantityOrdered: number; unitPrice: number;
}
export interface PurchaseOrderRequest {
  supplierId: number; orderedBy: number;
  notes: string; expectedDate: string;
  items: PurchaseOrderItemRequest[];
}
export interface ApproveRequest { approvedBy: number; }
export interface ReceiveGoodsRequest { performedBy: number; }
